### PR TITLE
fix: release workflow

### PR DESCRIPTION
**Pull request by ruben-arts**
With the coming of our new release workflow, we've hit some bugs with the installer. 

This fixes the fact that `install.sh` couldn't read the `.tar.gz`files with subdir anymore.
It should also not create a published release anymore, and make a draft.

Testing this is hard. When this is merged I can make a `dry-run` on main and test more parts.

### CODE REVIEW

**Comment by ruben-arts** [comment]
This also needs to be done for `pixi self-update` but we need a transition period for that.

**Review by nichmor — APPROVED** [review]
bash script look logical to me.
`release.yaml` also looks right and the `self_update.rs` code is tested.

from my side I don't see anything to add/change

File: Cargo.toml [hunk 1]
```diff
@@ -386,10 +386,12 @@ dispatch-releases = true
 github-release = "announce"
 # Whether to auto-include files like READMEs, LICENSEs, and CHANGELOGs (default true)
 auto-includes = false
+# Whether to put the binaries in the root of the archive or not, this only applies to unix archives
```
Comment (wolfv) [inline Cargo.toml:389]: Does that mean self-update on Windows does still not work?

Comment (ruben-arts) [reply Cargo.toml:389]: It was not a problem for the .zip files, there the files where always in the root of the file.

**Pull request merged**